wdl2cwl/main.py

```diff
@@ -229,6 +229,7 @@ def __init__(self) -> None:
         """Initialize the sets used by the object and prevent inconsistent behaviours."""
         self.non_static_values: Set[str] = set()
         self.optional_cwl_null: Set[str] = set()
+        self.scatter_names: List[str] = []
```

Comment:
I don't understand this list. Is this a list of tasks that got scattered, the source of the scatter, the destination, or the newly created variable made as a result of WDL's `scatter` construct?

Reply:
It serves as a stack for scatter objects as they are encountered.

Reply:
When a scatter input is encountered it's destination is stored in the stack. The destination is stored is stored because we don't have access to the source withing the input object.

Reply:
What if there is a sub-workflow that has a scatter, would it conflict?

What about a workflow that has two separate scatters, what happens then?

Maybe another data structure that relates the scatter call name, the source, and the destination together would be safer?